Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -581,13 +586,13 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
             do {
                 try await tokenHandler.getToken()
             } catch {
-                throw TunnelError.startingTunnelWithoutAuthToken
+                throw TunnelError.startingTunnelWithoutAuthToken(internalError: error)
             }
         case .reset:
             // This case should in theory not be possible, but it's ideal to have this in place
             // in case an error in the controller on the client side allows it.
-            try await tokenHandler.removeToken()
-            throw TunnelError.startingTunnelWithoutAuthToken
+            try? await tokenHandler.removeToken()
```

Comment:
Even if this fails, this isn't the real failure reason - the real failure reason is the token is being removed which, as described above, is an abnormal scenario.  The removal failing shouldn't be the error we get.